Lib/functools.py

```diff
@@ -236,7 +236,7 @@ def __ge__(self, other):
 
 _initial_missing = object()
 
-def reduce(function, sequence, initial=_initial_missing):
+def reduce(function, sequence, initial=_initial_missing, /):
```

Comment:
This might be a breaking change for some project that reuse our stdlib. Like RustPython, PyPy, etc

Reply:
That might be a breaking change, but unlike PyPy, RustPython is a small project, and we can quickly moving forward.
Actually, every python release practically caused breaking change for RustPython.

About this case, @Eclips4 found RustPython has `_functools` in stdlib. So python version looks like not being used.

Reply:
> This might be a breaking change for some projects that reuse our stdlib. Like RustPython, PyPy, etc

They should not accept keyword arguments to follow the default functools.reduce() behavior.

Reply:
FYI pypy accepts keyword arguments:

```
$ pypy3.9
>>>> import functools
>>>> functools.reduce(function=str, sequence=("abc",))
'abc'
```

PyPy has no _functools extension.

Reply:
cc @cfbolz as PyPy maintainer